ee/clickhouse/sql/clickhouse.py

```diff
@@ -36,9 +42,13 @@
 , _offset UInt64
 """
 
+COLLAPSING_MERGE_TREE = "collapsing_merge_tree"
+
 
-def table_engine(table: str, ver: Optional[str] = None) -> str:
-    if ver:
+def table_engine(table: str, ver: Optional[str] = None, engine_type: Optional[str] = None) -> str:
+    if engine_type == COLLAPSING_MERGE_TREE and ver:
+        return COLLAPSING_TABLE_ENGINE.format(table=table, ver=ver)
+    elif ver:
         return TABLE_ENGINE.format(table=table, ver=ver)
```

Comment:
Perhaps worth renaming this to `REPLACING_TABLE_ENGINE` or smth + make sure we pass in engine_type in this branch as well? :)

Reply:
yup just update those and made sure engine_type is passed in everywhere